crates/btcio/src/reader/query.rs

```diff
@@ -105,6 +110,68 @@ async fn do_reader_task<R: Reader>(
     }
 }
 
+/// Checks for epoch and scan rule update. If so, rescan recent blocks to account for possibly
+/// missed relevant txs.
+async fn check_and_update_epoch_scan_rules_change<R: Reader>(
+    ctx: &ReaderContext<R>,
+    state: &mut ReaderState,
+) -> anyhow::Result<()> {
+    let latest_epoch = ctx.status_channel.epoch().unwrap_or(0);
```

Comment:
`last_epoch` is more precise as "latest" implies it's what `curr_epoch` would be.